WIP: Additional analysis for multi-component Euler systems.

SimonCan opened the pull request:
This adds analysis options, like enstrophy for the multicomponent Euler scheme.

SimonCan committed:
Added steps to compute the gradient for hyperbolic systems.

SimonCan committed:
Moved potential gradient calculations into integration routine.

DanielDoehring commented on src/equations/compressible_navier_stokes_2d.jl:
```diff
@@ -288,14 +288,14 @@ end
 
-@inline function enstrophy(u, gradients, equations::CompressibleNavierStokesDiffusion2D)
+@inline function enstrophy(u, equations::CompressibleNavierStokesDiffusion2D; gradients)
     # Enstrophy is 0.5 rho ω⋅ω where ω = ∇ × v
 
     omega = vorticity(u, gradients, equations)
     return 0.5f0 * u[1] * omega^2
 end
 
-@inline function vorticity(u, gradients, equations::CompressibleNavierStokesDiffusion2D)
+@inline function vorticity(u, equations::CompressibleNavierStokesDiffusion2D; gradients)
```
Why different signature compared to old and added functions?

SimonCan committed:
Distinguish between call using functions with 2 or 3 parameters.

SimonCan committed:
Reverted function signature.

SimonCan committed:
Update src/equations/compressible_euler_multicomponent_2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/equations/compressible_euler_multicomponent_2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/equations/compressible_euler_multicomponent_2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Added Andrew's Fortran code for derivatives.
Attempt of using existing infrastructure for parabolic terms to compute derivatives.

SimonCan committed:
Update src/callbacks_step/analysis_dg2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/callbacks_step/analysis_dg2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/callbacks_step/analysis_dg2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/semidiscretization/semidiscretization.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Sterted conversion of Andrew's derivatives method.

SimonCan committed:
Fuirther implementations of Andrew's derivative funciton.

SimonCan committed:
Further translations of Fortran variables to Trixi.jl variables.

SimonCan committed:
Added calculation of fluxes by direciton.

SimonCan committed:
Corrected calculation of gradients.

SimonCan committed:
Corrected the gradient function.

SimonCan committed:
Added 'semi' option to function call.

SimonCan committed:
Take care of cases when function is cons2cons.

SimonCan committed:
Update src/callbacks_step/analysis.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/callbacks_step/analysis.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/callbacks_step/analysis_dg2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Removed redundant 'semi' parameter.

SimonCan committed:
Update src/callbacks_step/analysis_dg2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/callbacks_step/analysis_dg2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/callbacks_step/analysis_dg2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/callbacks_step/analysis_dg2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/callbacks_step/analysis_dg2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Update src/callbacks_step/analysis_dg2d.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

SimonCan committed:
Removed redundant code.

SimonCan committed:
Merge branch 'sc/analysis_multi_euler_2d' of https://github.com/trixi-framework/Trixi.jl into sc/analysis_multi_euler_2d

SimonCan committed:
Remoevd some redundant code.

SimonCan committed:
Compute now also the y-derivative.

SimonCan committed:
Use components of derivative vector for computing the vorticity.

SimonCan committed:
Merge branch 'main' into sc/analysis_multi_euler_2d

SimonCan committed:
Removed factor rho from the enstrophy.

SimonCan committed:
Replaced DG derivative with polynoial interpolation.

SimonCan committed:
Merge branch 'main' into sc/analysis_multi_euler_2d

SimonCan committed:
Merge remote-tracking branch 'origin/main' into sc/analysis_multi_euler_2d

SimonCan committed:
Added max_abs_speed_naive for CompressibleEulerMulticomponentEquations2D
and normal_direction::AbstractVector.

SimonCan committed:
Applied auto formatter.
